web3/_utils/module_testing/eth_module.py

```diff
@@ -196,6 +196,23 @@ def test_eth_sign(self, web3, unlocked_account_dual_type):
         )
         assert new_signature != signature
 
+    def test_eth_signTransaction(self, web3, unlocked_account, geth_signed_tx=None):
+        txn_params = {
+            'from': unlocked_account,
+            'to': unlocked_account,
+            'value': 1,
+            'gas': 21000,
+            'gasPrice': web3.eth.gasPrice,
+            'nonce': 0,
+        }
+        COINBASE_PK = '0x58d23b55bc9cdce1f18c2500f40ff4ab7245df9a89505e9b1fa4851f623d241d'
+        result = web3.eth.signTransaction(txn_params)
+        actual = web3.eth.account.signTransaction(txn_params, COINBASE_PK)
```

Comment:
This may be a dumb question, but why do we need both `web3.eth.signTransaction` and `web3.eth.account.signTransaction`?

Reply:
Not dumb at all, it could very well be a dumb idea. My thinking was that `web3.eth.account.signTransaction` (aka [`Account.signTransaction` from `eth-account`](https://github.com/ethereum/eth-account/blob/master/eth_account/account.py#L397) does pretty much the same thing as sending a `eth_signTransaction` JSONRPC request to a client, which is what `web3.eth.signTransaction` does. Since the eth-account implementation was well-done, it seemed to me a fair test comparison to make sure that `web3.eth.signTransaction` is performing as expected

Reply:
Yup, pretty much the only difference is whether your private key is managed by your python app or your node.

Reply:
Got it! That was the missing link 🔗. Thanks!

Reply:
No dumb questions.... only dumb people :smiling_imp: